packages/components/src/scroll-to-top/scroll-to-top-on-mount.tsx

```diff
@@ -0,0 +1,10 @@
+import { useEffect } from 'react';
+
+export function ScrollToTopOnMount() {
```

Comment:
It would be a bummer not to reuse this from somewhere else. It's been done many times, and here we're introducing a new one without attempting to reuse / refactor the original ones.

Here are a few examples of similar prior art I found on my first search:

https://github.com/Automattic/wp-calypso/blob/f2e6f88ae65c8571bbac21b284c29b48ec987598/client/reader/full-post/controller.js#L8

https://github.com/Automattic/wp-calypso/blob/f2e6f88ae65c8571bbac21b284c29b48ec987598/client/reader/full-post/controller.js#L8

https://github.com/Automattic/wp-calypso/blob/7e6d2b9c30257d9270c1c3aa8db0168945cec8a4/client/my-sites/plugins/controller.js#L141

https://github.com/Automattic/wp-calypso/blob/d648de9f1b6ebfa95d7729e1c38fb9f0ef61033e/client/my-sites/picker/picker.jsx#L74

I generally don't have very strong feelings about this, but just imagine if every time we need something so generic, we reinvent the wheel and create a new solution. We'll end with dozens of them, and the cumulative effect of that practice can impact Calypso negatively, from a maintenance, performance, and developer experience perspective.

Reply:
@tyxla, Thank you for your in-depth review. I really appreciate it.
Before continuing with this PR, I'll reconsider a new approach to fix this issue. I'll make sure to tag you when I have something solid in mind.
I'll keep in mind reusing the existing resources and trying to fix it holistically.
Thank you!